ci/sync-gateway-crds.sh

```diff
@@ -0,0 +1,119 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# Function to source specific variables from the setup-kind.sh script
+source_specific_vars() {
+    local file="./ci/kind/setup-kind.sh"
+    local vars_to_source=("CONFORMANCE_VERSION" "CONFORMANCE_CHANNEL")
+    
+    for var in "${vars_to_source[@]}"; do
+        eval $(grep "^$var=" "$file")
+    done
+}
+
+# Source the required variables
+source_specific_vars
+
+# Capitalize the first letter of CONFORMANCE_CHANNEL
+CAPITALIZED_CHANNEL="$(echo "${CONFORMANCE_CHANNEL:0:1}" | tr '[:lower:]' '[:upper:]')${CONFORMANCE_CHANNEL:1}"
+
+# Define output directory and filenames
+OUT_DIR="${OUT_DIR:-projects/gateway2/crds}"
+OUT_FILENAME="${OUT_FILENAME:-gateway-crds.yaml}"
+TCPROUTE_FILENAME="tcproute-crd.yaml"
+
+# Create the output directory if it doesn't exist
+mkdir -p "${OUT_DIR}"
+
+# URLs for CRDs
+GATEWAY_CRD_URL="https://github.com/kubernetes-sigs/gateway-api/releases/download/${CONFORMANCE_VERSION}/${CONFORMANCE_CHANNEL}-install.yaml"
+TCPROUTE_CRD_URL="https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/refs/tags/${CONFORMANCE_VERSION}/config/crd/experimental/gateway.networking.k8s.io_tcproutes.yaml"
+
+# Header to prepend to the TCPRoute CRD file
+HEADER=$(cat <<EOF
+# Copyright 2024 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#
+# Gateway API ${CAPITALIZED_CHANNEL} channel install
+#
+---
+EOF
+)
+
+# Function to compare files ignoring headers (for TCPRoute only)
+compare_files_no_header() {
+    local file1="$1"
+    local file2="$2"
+
+    # Strip the header from both files and compare the rest of the content
+    tail -n +$(($(echo "$HEADER" | wc -l) + 1)) "$file1" > "$file1.stripped"
+    tail -n +$(($(echo "$HEADER" | wc -l) + 1)) "$file2" > "$file2.stripped"
+    cmp -s "$file1.stripped" "$file2.stripped"
+    local result=$?
+    rm -f "$file1.stripped" "$file2.stripped"
+    return $result
+}
+
+download_gateway_crd() {
+    local url="$1"
+    local dest="$2"
+
+    echo "Downloading Gateway CRD from $url to $dest..."
+    curl -sLo "$dest.tmp" "$url"
+    
+    if [ -f "$dest" ] && cmp -s "$dest" "$dest.tmp"; then
+        echo "No changes detected in $dest."
+        rm "$dest.tmp"
+    else
+        mv "$dest.tmp" "$dest"
+        echo "Updated $dest."
+    fi
+}
+
+download_tcproute_crd() {
```

Comment:
if i delete the existing crds, then run `make sync-gateway-crds`, i get an error:
```
rm: projects/gateway2/crds/tcproute-crd.yaml.tmp.full: No such file or directory
make: *** [sync-gateway-crds] Error 1
```

Reply:
Resolved by commit https://github.com/solo-io/gloo/commit/4110d13ce31ca34133775fd02fd54a4bbcd5350a